syntax = "proto3";

package MATNew;

message nucMut
{
    int32 nucPosition = 1; 
    int32 nucGapPosition = 2;  
    bool  nucGapExist = 3;
    int64 blockId = 4;
    bool  blockGapExist = 5;
    uint32 mutInfo = 6; 
}

message blockMut
{
    int64 blockId = 1;
    bool  blockGapExist = 2;
    bool  blockMutInfo = 3;
}

message node 
{
    repeated nucMut   nucMutation = 1;
    repeated blockMut blockMutation = 2;
    repeated string   annotations = 3;
}


message block
{
    int64            blockId = 1;
    bool             blockGapExist = 2;
    repeated uint32  consensusSeq = 3;
    string           chromosomeName = 4;
}

message gapList
{
    repeated int32 nucPosition = 1;
    int64          blockId = 2;
    bool           blockGapExist = 3;
    repeated int32 nucGapLength = 4;
}

message blockGapList
{
    repeated int32  blockPosition = 1;
    repeated int32  blockGapLength = 2;
}

message tree 
{
    string           newick = 1;
    repeated node    nodes = 2;
    repeated block   blocks = 3;
    repeated gapList gaps = 4;
    blockGapList     blockGaps = 5;
}